library/ssl_tls13_generic.c

```diff
@@ -1199,6 +1200,19 @@ int mbedtls_ssl_reset_transcript_for_hrr( mbedtls_ssl_context *ssl )
 #if defined(MBEDTLS_SHA256_C) || defined(MBEDTLS_SHA384_C)
     ssl->handshake->update_checksum( ssl, hash_transcript, hash_len );
 #endif /* MBEDTLS_SHA256_C || MBEDTLS_SHA384_C */
+
+    /* Destroy generated private key. */
+    status = psa_destroy_key( ssl->handshake->ecdh_psa_privkey );
```

Comment:
@mprse @mpg @xffbai I think this should be in 'ssl_tls13_reset_key_share` which was not changed by this PR and still call  `mbedtls_ecdh_free()`. Do I miss something ?

Reply:
I don't think you're missing something, but we probably did.

Reply:
The next question is: who should fix that? Does it fit in one of your PRs or should we open one for that?

Reply:
I am currently fixing it in the version negotiation PR. I am ok to move that to a specific PR if we think it is better.

Reply:
No, I think if it fits in an existing PR that's probably the easiest.

Reply:
I can create a fix now if needed.